### PR TITLE
Layer overhaul

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request
Changes up some layer and plane defines for no particular reason lol

## Why It's Good For The Game
Planes actually override layers, and layers control ordering within planes. A lot of the usage of plane and layer was wholly unnecessary. This refactor helps future maintainability while also being needed staging for _future features._

### CODE REVIEW

File: code/__DEFINES/layers.dm [hunk 1]
```diff
@@ -84,76 +91,90 @@
...
+#define EMISSIVE_BLOCKER_RENDER_TARGET "*EMISSIVE_BLOCKER_PLANE"
 
-#define FULLSCREEN_PLANE 31
-#define FLASH_LAYER 31
-#define FULLSCREEN_LAYER 31.1
-#define UI_DAMAGE_LAYER 31.2
-#define BLIND_LAYER 31.3
-#define CRIT_LAYER 31.4
-#define CURSE_LAYER 31.5
+///This plane is "unblockable" emissives. It does the same thing as the emissive plane but isn't masked by the smissive blocker plane. Use for on-mob and movable emissives.
```
Comment (Ghommie) [inline code/__DEFINES/layers.dm:139]:
```suggestion
///This plane is "unblockable" emissives. It does the same thing as the emissive plane but isn't masked by the emissive blocker plane. Use for on-mob and movable emissives.
```
e.

File: code/_onclick/hud/fullscreen.dm [hunk 2]
```diff
@@ -149,7 +149,7 @@
 	screen_loc = "WEST,SOUTH to EAST,NORTH"
 	icon_state = "flash"
 	plane = SPLASHSCREEN_PLANE
-	layer = SPLASHSCREEN_LAYER - 1
+	layer = -1
```
Comment (Ghommie) [inline code/_onclick/hud/fullscreen.dm:152]: Uh? Shouldn't this be a define?

File: code/modules/lighting/emissive_blocker.dm [hunk 3]
```diff
@@ -9,7 +9,7 @@
 /atom/movable/emissive_blocker
 	name = ""
 	plane = EMISSIVE_BLOCKER_PLANE
-	layer = EMISSIVE_BLOCKER_LAYER
+	layer = 0
```
Comment (Ghommie) [inline code/modules/lighting/emissive_blocker.dm:12]: Is this line even needed anymore?

**Review by Timberpoes — CHANGES_REQUESTED** [review]
```
DM compiler version 513.1536
loading tgstation.mdme
loading interface/skin.dmf
code/modules/shuttle/shuttle.dm:151:error: GHOST_LAYER: undefined var
tgstation.mdme.dmb - 1 error, 0 warnings (3/23/21 1:05 pm)
```

**Comment by actioninja** [comment]
Works on my machine

File: code/modules/lighting/emissive_blocker.dm (hunk 3 above)
Comment (actioninja) [reply code/modules/lighting/emissive_blocker.dm:12]: no, this was the result of a regex to find and replace these in add vis overlays.

File: code/_onclick/hud/fullscreen.dm (hunk 2 above)
Comment (actioninja) [reply code/_onclick/hud/fullscreen.dm:152]: It probably would be clearer as a define, sure.

**Comment by Mannybrado** [comment]
hi 🙂

**Comment by Timberpoes** [comment]
> 
> 
> Works on my machine

You missed `layer = GHOST_LAYER` in `/obj/docking_port/proc/highlight`. The reason this isn't happening on local is because this code is only included with `#ifdef TESTING`

If you #define TESTING on local you'll notice the error.

**Comment by actioninja** [comment]
That was a joke but I see

File: code/world.dm [hunk 4]
```diff
@@ -19,6 +19,7 @@
 	hub_password = "kMZy3U5jJHSiBQjr"
 	name = "/tg/ Station 13"
 	fps = 20
+	map_format = SIDE_MAP
```
Comment (Cyberboss) [inline code/world.dm:22]: http://www.byond.com/docs/ref/#/world/var/map_format
image

**Comment by Cyberboss** [comment]
On Bagil

**Comment by optimumtact** [comment]
bad merge

**Pull request merged**